libraries/botbuilder-dialogs-adaptive/src/recognizers/regexRecognizer.ts

```diff
@@ -94,81 +91,67 @@ export class RegexRecognizer extends Recognizer implements RegexRecognizerConfig
                 }
             }
 
-            if (matches.length > 0) {
-                const intentKey = intentPattern.intent.split(' ').join('_');
-                if (!recognizerResult.intents.hasOwnProperty(intentKey)) {
-                    recognizerResult.intents[intentKey] = { score: 1.0 };
-                    recognizerResult.intents[intentKey]['pattern'] = intentPattern.pattern;
-                }
+            if (matches.length === 0) {
+                continue;
+            }
 
-                for (let j = 0; j < matches.length; j++) {
-                    const match = matches[j];
-                    if (match.groups) {
-                        for (const name in match.groups) {
-                            const text = match.groups[name];
-                            if (text) {
-                                const entity: Entity = {
-                                    type: name,
-                                };
-                                entity['text'] = text;
-                                entity['start'] = match.index;
-                                entity['end'] = match.index + text.length;
-                                entityPool.push(entity);
-                            }
-                        }
-                    }
+            // TODO length weighted match and multiple intents
+            const intentKey = intentPattern.intent.split(' ').join('_');
+            recognizerResult.intents[`${intentKey}`] ??= {
+                score: 1.0,
+                pattern: intentPattern.pattern,
+            };
+
+            matches.forEach((match: RegExpExecArray) => {
+                if (match.groups) {
+                    Object.entries(pickBy(match.groups)).forEach(([name, text]) => {
```

Comment:
`pickBy` isn't really necessary is it?

Reply:
`pickBy` was used to filter out empty values from matched groups, it is necessary since values are always not empty.